### PR TITLE
Add bunnymark-style benchmark

### PR DESCRIPTION
**Pull request by franciscod**
Reference results on M2 mac mini:
image

(see below for updated results)

### CODE REVIEW

**Comment by franciscod** [comment]
On #36, there are three proposed variants for Bunnymark-style benchmarks:

- 🟥CPU🟥 [BunnyMark]: nodes: Run a bunnymark style benchmark with 5000 nodes.
- 🟥CPU🟥 [BunnyMark]: CanvasItem: Run a bunnymark style benchmark with 5000 bunnies drawn using CanvasItem API
- 🟥CPU🟥 [BunnyMark]: MeshInstance2D: Run a bunnymark style benchmark with 5000 bunnies drawn using MeshInstance2D, drawing directly into the 2D array.

This initial commit would cover the first variant. Does it look okay?
I'm working on CanvasItem API variant now, I can add those commits on this branch as well.

**Comment by franciscod** [comment]
image

**Comment by franciscod** [comment]
The three variants are ready.

Reference results on M2 mac mini:
image

File: benchmarks/rendering/bunnymark.gd [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+
+func benchmark_bunnymark_sprite2d_05_000() -> Node2D:
+	return Bunnymark.new({ count = 5000 })
+
+func benchmark_bunnymark_sprite2d_10_000() -> Node2D:
+	return Bunnymark.new({ count = 10000 })
+
+func benchmark_bunnymark_sprite2d_20_000() -> Node2D:
+	return Bunnymark.new({ count = 20000 })
```
Comment (Calinou) [inline benchmarks/rendering/bunnymark.gd:119]: For consistency with other benchmarks, I'd rename the names to avoid number separators and leading zeroes.

**Comment by Calinou** [comment]
Should be good to merge after applying suggestions.

Note that the CPU/GPU timings reported here don't quite reflect the actual performance level for some reason. I get a reported CPU/GPU time of about 3 ms in `bunnymark_sprite2d_20_000`, but I only get 90 FPS (= 11.1 ms).

**Comment by stuartcarnie** [comment]
I ran bunnymark on a MacBook Pro M1 Max comparing the Metal rendering device vs Vulkan, and we are seeing some very nice speedups. I still have some issues with the timestamp code in Metal that I am currently working on, so I've just used `--print-fps` for the `20_000` bunnies.

| benchmark | Metal FPS | Vulkan FPS |
|:---- |:---- |:---- |
| canvasitem_draw_api_20_000 | 97 (10.30 mspf) | 66 (15.15 mspf) |
| meshinstance2d_20_000 | 52 (19.23 mspf) | 38 (26.31 mspf) |
| sprite2d_20_000 | 53 (18.86 mspf) | 38 (26.31 mspf) |

**Review by Calinou — APPROVED** [review]
I applied the suggested renames.

Thanks!

**Comment by franciscod** [comment]
Thank you!

**Pull request merged**